src/Emulator/Visuals/Visuals.js

```diff
@@ -1,11 +1,236 @@
+  function plusClick() {
+    const zoomAmount = .1;
+    setZoom(zoom => zoom + zoomAmount);
+    setContentDimensions({
+      height: contentDimensions.height + additionalImageSpace,
+      width: contentDimensions.width + additionalImageSpace,
+    });
+    setElementBounds({
+      top: contentDimensions.height + additionalImageSpace,
+      bottom: contentDimensions.width + additionalImageSpace,
+      left: '',
+      right: contentDimensions.width + additionalImageSpace
+    })
+  }
+
+  // Zoom out function and adjusts drag handler size (the invisible space you can click and start drag with)
+  function minusClick() {
+    const zoomAmount = .1;
+    setZoom(zoom => zoom - zoomAmount);
+    setContentDimensions({
+      height: contentDimensions.height - additionalImageSpace,
+      width: contentDimensions.width - additionalImageSpace,
+    });
+    setElementBounds({
+      top: contentDimensions.height + additionalImageSpace,
+      bottom: contentDimensions.width + additionalImageSpace,
+      left: '',
+      right: contentDimensions.width + additionalImageSpace
+    })
+  }
+
+  // Full screen function; changes element resolution to container's max dimensions when when maxRes and toggleMax state changes
+  useEffect(() => {
+    if (toggleMax === true) {
+      setContentDimensions({
+        height: maxRes.height,
+        width: maxRes.width,
+      });
+    } else {
+      setContentDimensions({
+        height: maxRes.height,
+        width: maxRes.width,
+      });
+    }}, [maxRes, toggleMax]);
+
+  // Full screen function; finds cotainer and container space resolution sizes and sets them so that image fills the whole container using true/false states
+  function maxClick() {
+    setMax(!toggleMax);
+    if (toggleMax === false) {
+      setMaxRes({
+        height: containerDimensions.height || '49vh',
+        width: containerDimensions.width || '74vh',
+        prevHeight: contentDimensions.height,
+        prevWidth: contentDimensions.width,
+      });
+      setPos({
+        x: 0, y: 0,
+      });
+    } else {
+      const centerX = 150;
+      const centerY = 150;
+      setMaxRes({
+        height: maxRes.prevHeight,
+        width: maxRes.prevHeight,
+        prevHeight: maxRes.height,
+        prevWidth: maxRes.width,
+      });
+      setPos({
+        x: centerX, y: centerY,
+      });
+    }
+  }
+
+  function hideClick() {
+    return (
+        setHideImage(!hideImage)
+    );
+  }
+
   return (
-    <div className="visuals">
-      I am the Visuals Panel!
+  <div>
+    {/* Sets container size, space you can drag in container, and the element inside the container's size */}
+    <div className="imageContainer" style={{height: containerHeight, width: containerWidth, overflow: 'auto', padding: '0px'}}>
+      <div className="imageSpace" style={{height: containerDimensions.height || '49vh', width: containerDimensions.width || '74vw'}}>
+        {/* Handles drag */}
```

Comment:
what's this?

Reply:
The way I built the image panel was in 3 layers: the image container, the image space, and the draggable container. Here's an [image](https://i.imgur.com/9D9wKLR.png) that hopefully makes more sense:
In the new PR, you will see what I'm talking about visually now that there's a grey background over the drag handler.